From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 16 Apr 2026 18:38:22 +0200
Subject: mpris: Do not send player-removed for a player that has never been
 added

We only emit player-added for players that can play, but we remove them
all, despite this state.

This causes a non-fatal warning during the removal of a non-existent
notification (that is in fact an undefined message)

Closes: https://gitlab.gnome.org/GNOME/gnome-shell/-/work_items/8284

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4183
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/2144732
---
 js/ui/mpris.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/ui/mpris.js b/js/ui/mpris.js
index 44efbd6..c86d980 100644
--- a/js/ui/mpris.js
+++ b/js/ui/mpris.js
@@ -244,7 +244,8 @@ export const MprisSource = GObject.registerClass({
             if (player) {
                 this._players.delete(name);
                 player.disconnectObject(this);
-                this.emit('player-removed', player);
+                if (player.canPlay)
+                    this.emit('player-removed', player);
             }
         }
 
